pkg/server/server.go

```diff
@@ -273,9 +274,8 @@ func (s *Server) HTTPHandler() (http.Handler, error) {
 			os.Exit(1)
```

Comment:
not new, but could you please update the error handling similarly as you did below?

Reply:
As discussed, we can do this as a follow-up.

In general there are two different cases, errors from which we can recover vs. error from which we cannot recover and therefore we exit the process.